fix: misc fixes (integer primary keys)

phot0n opened the pull request:
Changes:
- `local.autoincremented_status_map` in `frappe.init` get resetted on every request (i think due to `init_request` method? it keeps on triggering `frappe.init` function) so switched to a simple dict
- simplified `is_val_used` in `set_next_val` function for sequences
- use multisql
- fixed fields not being refreshed (ref: https://github.com/frappe/frappe/pull/15964#issuecomment-1070352518)
- refactored `cast_name` method (and made it a simple function as opposed to class method)
- added `cast_name` wherever it was missed (hopefully 🤞)
- added sequence function to Database class - can now be referred/used using `frappe.db.x`
- fixed wrong query strings for `minvalue` and `maxvalue` for sequences
- improved regex in `modify_query` for postgres
- refactored `validate_autoname` method into a simple function
- allowed autoincrement doctype to change name column type if no data is present
- made autoincrement autoname to be set using customize form

TODO:
- [x] add tests related to sequences
- [x] fix tests
- [x] replace all imported sequence functions with `frappe.db.x`
- [x] move out casting from `extract_tables` into a separate function

phot0n committed:
fix: misc fixes

local.x gets resetted on every request so switched to a simple dict

simplified is_val_used in set_next_val function for sequences

phot0n committed:
chore: use multisql for sequence methods

phot0n committed:
fix: fields not updating on form

* minor(base_input): removed unnecessary branching in update_input

phot0n committed:
chore: remove prints and rename autoincremented_status_map

phot0n committed:
chore: added proper type hint + comment + formatting

phot0n committed:
chore: fix sider

* minor: testing template

phot0n commented on frappe/public/js/frappe/form/controls/base_input.js:
```diff
@@ -65,11 +65,7 @@ frappe.ui.form.ControlInput = class ControlInput extends frappe.ui.form.Control
-			} else {
-				me.set_input(me.value || null);
-			}
+			me.set_input(me.value);
```
removed unnecessary branching

phot0n commented:
failing test is unrelated (quite possibly related to https://github.com/frappe/frappe/commit/93a36092fda2813b4ecede33371a238999b1edd5)

![Screenshot 2022-03-24 at 11 22 22 AM](https://user-images.githubusercontent.com/32034600/159880046-1f9f1ae8-d127-4844-aca3-8338940b638d.png)

phot0n committed:
fix: added searching in cast_name rather than handling it manually

phot0n committed:
fix: share condition query + test_build_match_conditions

phot0n committed:
fix: add cast_name to more places

phot0n committed:
test: test for sequence

* fix: sequence functions

phot0n committed:
fix: inherit frappetestcase

phot0n committed:
chore: fix sider

phot0n committed:
minor: attach sequence methods to db context local

phot0n committed:
chore: update sequence function names in Database

use frappe.db for sequences in naming.py

phot0n commented:
failing tests  are not related

phot0n committed:
refactor: cast_name method

phot0n committed:
fix: convert filename to str (for autoincremented doctypes)

phot0n committed:
chore: better regex for modifying values for postgres

rmehta commented on frappe/model/db_query.py:
```diff
@@ -845,6 +811,18 @@ def update_user_settings(self):
 
 		update_user_settings(self.doctype, user_settings)
 
+def cast_name(column: str) -> str:
```
add a comment / docstring

rmehta commented on frappe/model/naming.py:
```diff
@@ -14,6 +13,8 @@
 	from frappe.model.meta import Meta
 
 
+autoincremented_site_status_map = {}
```
global variable?

phot0n replied:
yep ..tried the `frappe.local` objects but they seem to get reset on every request. Another option is to use redis but since this will be used heavily, just used a simple dictionary.

It's basically used to store the status of site, whether `log_types` are autoincremented for the site or not.

Open to any other suggestions.

phot0n committed:
minor: allow changing name column type (if no data is present in the doctype)

phot0n committed:
refactor: validate_autoname

converted it to a simple function
enabled changing autoincrement autoname from customize form

phot0n committed:
fix: use sql_ddl for change_column_type in postgres

phot0n committed:
fix: use not null constraint in postgres when changing name type

phot0n committed:
fix(test): updated test_autoincremented_doctype_transition with transitioning when no data is present

phot0n committed:
fix(test): updated test_cast_name

probably messed up during rebase

phot0n committed:
fix(test): used rollback upon error in transaction for postgres

* chore: use frappe.db.x methods for sequences

* minor: use temporary sequences in test

* minor: use generate_hash for sequence naming in sequence tests

phot0n committed:
chore: replace sequence imports with frappe.db.x

phot0n committed:
chore: move out casting name fields to a separate method

phot0n committed:
refactor: cast_name

more explicit cases for casts and added docstring

* fix: added space in test_cast_name

phot0n committed:
chore: fix linter

phot0n committed:
chore: better naming for can_change_name_column_type

phot0n committed:
chore: add comment for autoincremented_site_status_map

phot0n commented:
failing UI test is of `Form Tour` - seems unrelated

phot0n committed:
chore: update/add docstrings
